app/controllers/shipit/deploys_controller.rb

```diff
@@ -40,6 +41,16 @@ def revert
       redirect_to rollback_stack_deploy_path(@stack, previous_deploy)
     end
 
+    def rollback_until_commit(task)
+      if previous_successful_deploy(task)
+        @rollback_until_commit ||= Commit.where(
+          stack_id: @previous_successful_deploy.stack_id,
+          id: @previous_successful_deploy.until_commit_id,
+        ).first
```

Comment:
could you do `@previous_successful_deploy.until_commit` here?

Reply:
I think that'd work, but I'm replacing this method altogether now.